unable to build project

when running `./manage.py assets build` I am running into:

```
$ ./manage.py assets build
Generating bundle: js/ember-app-%(version)s.js
Traceback (most recent call last):
  File "./manage.py", line 6, in <module>
    manager.run()
  File "/usr/lib/python2.7/site-packages/flask_script/__init__.py", line 423, in run
    result = self.handle(sys.argv[0], sys.argv[1:])
  File "/usr/lib/python2.7/site-packages/flask_script/__init__.py", line 402, in handle
    return handle(app, *positional_args, **kwargs)
  File "/usr/lib/python2.7/site-packages/flask_script/commands.py", line 145, in handle
    return self.run(*args, **kwargs)
  File "/home/jharting/projects/skylines/skylines/commands/assets.py", line 17, in build
    bundle.build()
  File "/usr/lib/python2.7/site-packages/webassets/bundle.py", line 563, in build
    disable_cache=disable_cache))
  File "/usr/lib/python2.7/site-packages/webassets/bundle.py", line 498, in _build
    force, disable_cache=disable_cache, extra_filters=extra_filters)
  File "/usr/lib/python2.7/site-packages/webassets/bundle.py", line 355, in _merge_and_apply
    resolved_contents = self.resolve_contents(env, force=True)
  File "/usr/lib/python2.7/site-packages/webassets/bundle.py", line 148, in resolve_contents
    raise BundleError(e)
webassets.exceptions.BundleError: 'ember/assets/skylines.js' not found in load path: ['/home/jharting/projects/skylines/skylines/frontend/static']
```

Happens both when running in my local environment and also in vagrant. Any idea what I am doing wrong?

I'm trying to fix this issue, unfortunately I'm stuck with the following error message:
```
ubuntu@ubuntu-xenial:/vagrant/ember$ ember build  
Cannot find module './code-path-analysis/code-path-analyzer'
Error: Cannot find module './code-path-analysis/code-path-analyzer'
    at Function.Module._resolveFilename (module.js:326:15)
    at Function.Module._load (module.js:277:25)
    at Module.require (module.js:354:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/vagrant/ember/node_modules/ember-cli-eslint/node_modules/broccoli-lint-eslint/node_modules/eslint/lib/eslint.js:20:24)
    at Module._compile (module.js:410:26)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Module.require (module.js:354:17)
ubuntu@ubuntu-xenial:/vagrant/ember$ npm --version
3.5.2
ubuntu@ubuntu-xenial:/vagrant/ember$ nodejs --version
v4.2.6
```
My updated [Vagrantfile](https://github.com/kerel-fs/skylines/blob/fix/vagrant/Vagrantfile) is based on the official `Ubuntu/xenial64` base box and installs a working node.js v4 environment. Any ideas how to fix the error above?

Yes, `ember build` fails again with
```
vagrant@vagrant-ubuntu-precise-64:/vagrant/ember$ ./node_modules/ember-cli/bin/ember build
Cannot find module './code-path-analysis/code-path-analyzer'
Error: Cannot find module './code-path-analysis/code-path-analyzer'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/vagrant/ember/node_modules/ember-cli-eslint/node_modules/broccoli-lint-eslint/node_modules/eslint/lib/eslint.js:20:24)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
vagrant@vagrant-ubuntu-precise-64:/vagrant/ember$ npm --version
3.10.3
vagrant@vagrant-ubuntu-precise-64:/vagrant/ember$ node --version                              
v6.6.0
```

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -12,15 +12,6 @@ export POSTGIS_GDAL_ENABLED_DRIVERS=GTiff
 export POSTGIS_ENABLE_OUTDB_RASTERS=1
 EOF
 
-# adjust apt-get repository URLs
-
-sudo bash -c "cat > /etc/apt/sources.list" << EOF
-deb mirror://mirrors.ubuntu.com/mirrors.txt precise main restricted universe multiverse
-deb mirror://mirrors.ubuntu.com/mirrors.txt precise-updates main restricted universe multiverse
-deb mirror://mirrors.ubuntu.com/mirrors.txt precise-backports main restricted universe multiverse
-deb mirror://mirrors.ubuntu.com/mirrors.txt precise-security main restricted universe multiverse
-EOF
-
 sudo bash -c "cat > /etc/apt/sources.list.d/pgdg.list" << EOF
 deb http://apt.postgresql.org/pub/repos/apt/ precise-pgdg main
 EOF
@@ -89,8 +80,8 @@ sudo sudo -u postgres psql -d skylines_test -c 'CREATE EXTENSION fuzzystrmatch;'
 SCRIPT
 
 Vagrant.configure("2") do |config|
-  config.vm.box = 'ubuntu-12.04.2-x86_64'
-  config.vm.box_url = 'http://puppet-vagrant-boxes.puppetlabs.com/ubuntu-server-12042-x64-vbox4210.box'
+  # TravisCI uses a Precise Pangolin base image
+  config.vm.box = 'ubuntu/precise64'
 
   config.vm.network 'forwarded_port', guest: 5000, host: 5000
   config.vm.network 'forwarded_port', guest: 5001, host: 5001

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -50,29 +50,31 @@ sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-6 60 --slave /u
 wget -N -nv https://bootstrap.pypa.io/get-pip.py
 sudo -H python get-pip.py
 
-# install skylines frontend dependencies
-
-cd /vagrant/ember
+# install nvm, node 4.x, npm and bower
 
 curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.32.0/install.sh | bash
 
-# load nvm
 export NVM_DIR="/home/vagrant/.nvm"
 [ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"
 
-nvm install 6
-npm install
-npm install -g bower
-bower install
-
-# build skylines frontend
-
-ember build
+nvm install 4
+npm install -g bower ember-cli phantomjs-prebuilt
 
 # install skylines and the python dependencies
 
 cd /vagrant
 sudo -H pip install -r requirements.txt --no-binary greenlet
+
+# install skylines frontend dependencies and build it
+
+cd ember
+
+npm install
+bower install
+
+ember build
+
+cd ..
 
 # create PostGIS databases
 

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -50,6 +50,25 @@ sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-6 60 --slave /u
 wget -N -nv https://bootstrap.pypa.io/get-pip.py
 sudo -H python get-pip.py
 
+# install skylines frontend dependencies
+
+cd /vagrant/ember
+
+curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.32.0/install.sh | bash
+
+# load nvm
+export NVM_DIR="/home/vagrant/.nvm"
+[ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"
+
+nvm install 6
+npm install
+npm install -g bower
+bower install
+
+# build skylines frontend
+
+ember build
+
 # install skylines and the python dependencies
 
 cd /vagrant

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -104,6 +104,12 @@ Vagrant.configure("2") do |config|
   # TravisCI uses a Precise Pangolin base image
   config.vm.box = 'ubuntu/precise64'
 
+  # increase memory size, required by 'npm install'
+  config.vm.provider "virtualbox" do |v|
+    v.memory = 2048
+    v.cpus = 2
+  end
+
   config.vm.network 'forwarded_port', guest: 5000, host: 5000
   config.vm.network 'forwarded_port', guest: 5001, host: 5001
 

